src/Microsoft.PowerApps.TestEngine/Modules/TestEngineExtensionChecker.cs

```diff
@@ -288,6 +317,45 @@ public bool VerifyContainsValidNamespacePowerFxFunctions(TestSettingExtensions s
 
                 foreach (TypeDefinition type in module.GetAllTypes())
                 {
+                    // Provider checks are based on Namespaces string[] property
+                    if (
+                        type.Interfaces.Any(i => i.InterfaceType.FullName == "Microsoft.PowerApps.TestEngine.Providers.ITestWebProvider")
+                        ||
+                        type.Interfaces.Any(i => i.InterfaceType.FullName == "Microsoft.PowerApps.TestEngine.Users.IUserManager")
+                        )
+                    {
+                        if (CheckPropertyArrayContainsValue(type, "Namespaces", out var values))
+                        {
+                            foreach (var name in values)
+                            {
+                                // Check against deny list using regular expressions
+                                if (settings.DenyPowerFxNamespaces.Any(pattern => Regex.IsMatch(name, WildcardToRegex(pattern))))
+                                {
+                                    Logger.LogInformation($"Deny Power FX Namespace {name} for {type.Name}");
+                                    return false;
+                                }
+
+                                // Check against deny wildcard and allow list using regular expressions
+                                if (settings.DenyPowerFxNamespaces.Any(pattern => pattern == "*") &&
+                                    (!settings.AllowPowerFxNamespaces.Any(pattern => Regex.IsMatch(name, WildcardToRegex(pattern))) &&
+                                     name != "TestEngine"))
+                                {
+                                    Logger.LogInformation($"Deny Power FX Namespace {name} for {type.Name}");
+                                    return false;
+                                }
+
+                                // Check against allow list using regular expressions
+                                if (!settings.AllowPowerFxNamespaces.Any(pattern => Regex.IsMatch(name, WildcardToRegex(pattern))) &&
+                                    name != "TestEngine")
```

Comment:
If "TestEngine" is used multiple times, use a const.

Reply:
Updated